crates/subspace-farmer/src/single_disk_farm.rs

```diff
@@ -122,10 +158,291 @@ impl SingleDiskFarmMetadata {
 
     /// Store `SingleDiskFarm` metadata to path where metadata is supposed to be stored so it can be
     /// loaded again upon restart.
-    pub fn store_to(&self, metadata_path: &Path) -> io::Result<()> {
+    pub fn store_to(&self, metadata_directory: &Path) -> io::Result<()> {
         fs::write(
-            metadata_path.join(Self::FILE_NAME),
+            metadata_directory.join(Self::FILE_NAME),
             serde_json::to_vec(self).expect("Metadata serialization never fails; qed"),
         )
     }
+
+    // ID of the farm
+    pub fn id(&self) -> &SingleDiskFarmId {
+        let Self::V0 { id, .. } = self;
+        id
+    }
+
+    // Genesis hash of the chain used for farm creation
+    pub fn genesis_hash(&self) -> &[u8; 32] {
+        let Self::V0 { genesis_hash, .. } = self;
+        genesis_hash
+    }
+
+    // How much space in bytes can farm use for plots (metadata space is not included)
+    pub fn usable_plotting_space(&self) -> u64 {
+        let Self::V0 {
+            usable_plotting_space,
+            ..
+        } = self;
+        *usable_plotting_space
+    }
+
+    // IDs of single plot farms contained within
+    pub fn single_plot_farms(&self) -> &[SinglePlotFarmId] {
+        let Self::V0 {
+            single_plot_farms, ..
+        } = self;
+        single_plot_farms
+    }
```

Comment:
We can, however the reason these methods exist is so that we don't have to match on enum at all, we can call `.id()` and get ID regardless of what version it happens to be.

Reply:
Yeah, ikr. This what I imagined:
```rust
enum SingleDiskFarmMetadata {
    V0(SingleDiskFarmMetadataV0)
}

struct SingleDiskFarmMetadataV0 {
    // all fields are private
}
```